src/mlpack/core/data/dictionary_encoding_impl.hpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * @file dictionary_encoding_impl.hpp
+ * @author Jeffin Sam
+ *
+ * Implementation of dictionary encoding functions.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_DICT_ENCODING_IMPL_HPP
+#define MLPACK_CORE_DATA_DICT_ENCODING_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "dictionary_encoding.hpp"
+
+namespace mlpack {
+namespace data {
+
+void DicitonaryEncoding::Reset()
+{
+  mappings.clear();
+}
+
+template<typename TokenizerType>
+void DicitonaryEncoding::CreateMap(std::string& strings,
+    TokenizerType tokenizer)
+{
+  boost::string_view strView(strings);
+  boost::string_view token;
+  token = tokenizer(strView);
+  std::size_t curLabels = mappings.size() + 1;
```

Comment:
Why don't you use a label that equals to zero?

Reply:
In case where a user wanted to extend his map and introduce new mapping of words , if i start with 0 again it would override the previous one, for example :
```cpp
  data::DicitonaryEncoding en;
  en.createmap("hello world");
  en.createmap("bye world");
```
In the above case world will be mapped to 2, but hello and bye would be 0 each.

to rectify above case a user might use concatenation of the two strings.

Reply:
Also i have used 0 to pad the sp_mat or mat which will be taken as output and hence i can't use 0 and have to start from 1.

Reply:
I don't see the connection between the matrix and the minimum label. Could you elaborate a bit?

Reply:
suppose the strings are `Hello How ` and `Good Hello I` 
when encoded as `Hello - 0 , How -1 Good -2 and I - 3` 
the matrix will be 
```
 0 1 0
 2 0 3
```

But if we see , the padding and hello seems same ( the last 0, in first column is for padding ) and rest are encoding for hello, and thus we should encode hello as 1 and rest accordingly to get
```
1 2 0
3 1 4
```

Reply:
@jeffin143 Yes, you're right. Thanks for the clarification!